### PR TITLE
Add React cheatsheet

### PR DESCRIPTION
**Pull request by EugeneGohh**
## What does this PR do?
Add resource | Improve repo

### Description
Add Codecademy's `React` cheatsheet and fix some misunderstandings. I think it would be better not to mix `JavaScript` or `Python` with `HTML`.

### Why is this valuable (or not)?
Valuable for beginners who want to learn React.

### How do we know it's really free?
It's a free cheatsheet & available for download.

### For book lists, is it a book? For course lists, is it a course? etc.
Cheatsheet

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by davorpa** [comment]
Remember change draft status to ready for review, when you think is enough :wink:

Also check linter errors and try to fix it. List alphabetization or whitespaces are the most common mistakes

File: more/free-programming-cheatsheets.md [hunk 1]
```diff
@@ -105,6 +106,9 @@
 
 * [Vue Essential Cheatsheet](https://www.vuemastery.com/pdf/Vue-Essentials-Cheat-Sheet.pdf) - Vue Mastery (PDF)
 
+#### React.js
```
Comment (LuigiImVector) [inline more/free-programming-cheatsheets.md:109]:
```suggestion

#### React.js

```

There weren't the [correct number of black lines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#formatting).

Comment (EugeneGohh) [reply more/free-programming-cheatsheets.md:109]: Resolved

**Comment by EugeneGohh** [comment]
> Remember change draft status to ready for review, when you think is enough 😉
> 
> Also check linter errors and try to fix it. List alphabetization or whitespaces are the most common mistakes

I have tried to fix it, it seems like I can't get through the check.

File: more/free-programming-cheatsheets.md [hunk 2]
```diff
@@ -13,6 +13,7 @@
 * [JavaScript](#javascript)
   * [jQuery](#jquery)
   * [Vue.js](#vuejs)
+  * [React.js](#reactjs)
```
Comment (LuigiImVector) [inline more/free-programming-cheatsheets.md:16]:
```suggestion
  * [React.js](#reactjs)
  * [Vue.js](#vuejs)
```

The new category was not in alphabetical order.

Comment (EugeneGohh) [reply more/free-programming-cheatsheets.md]: Looks good now thanks a lot.

File: more/free-programming-cheatsheets.md [hunk 3]
```diff
@@ -68,10 +69,10 @@
...
 
 * [CSS CheatSheet](https://htmlcheatsheet.com/css/) (HTML)
-* [CSS Flexbox Cheatsheet](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) - Chris Coyier (HTML)
-* [CSS Grid Cheatsheet](https://css-tricks.com/snippets/css/complete-guide-grid/) - Chris House (HTML)
-* [FLEX: A simple visual cheatsheet for flexbox](https://flexbox.malven.co) - Chris Malven (HTML)
-* [GRID: A simple visual cheatsheet for CSS Grid Layout](https://grid.malven.co) - Chris Malven (HTML)
+* [CSS Flexbox Cheatsheet](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) - Chris Coyier (CSS)
+* [CSS Grid Cheatsheet](https://css-tricks.com/snippets/css/complete-guide-grid/) - Chris House (CSS)
+* [FLEX: A simple visual cheatsheet for flexbox](https://flexbox.malven.co) - Chris Malven (CSS)
+* [GRID: A simple visual cheatsheet for CSS Grid Layout](https://grid.malven.co) - Chris Malven (CSS)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:75]: Revert

```markdown
* [CSS Flexbox Cheatsheet](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) - Chris Coyier (HTML)
* [CSS Grid Cheatsheet](https://css-tricks.com/snippets/css/complete-guide-grid/) - Chris House (HTML)
* [FLEX: A simple visual cheatsheet for flexbox](https://flexbox.malven.co) - Chris Malven (HTML)
* [GRID: A simple visual cheatsheet for CSS Grid Layout](https://grid.malven.co) - Chris Malven (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 4]
```diff
@@ -90,9 +91,9 @@
 
 ### JavaScript
 
-* [JavaScript Cheatsheet](https://www.codecademy.com/learn/introduction-to-javascript/modules/learn-javascript-introduction/cheatsheet) - Codecademy (HTML)
-* [JavaScript CheatSheet](https://htmlcheatsheet.com/js/) (HTML)
-* [JavaScript Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/javascript) - Debuggex (HTML)
+* [JavaScript Cheatsheet](https://www.codecademy.com/learn/introduction-to-javascript/modules/learn-javascript-introduction/cheatsheet) - Codecademy (JavaScript)
+* [JavaScript CheatSheet](https://htmlcheatsheet.com/js/) (JavaScript)
+* [JavaScript Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/javascript) - Debuggex (JavaScript)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:96]: Revert
```markdown
* [JavaScript Cheatsheet](https://www.codecademy.com/learn/introduction-to-javascript/modules/learn-javascript-introduction/cheatsheet) - Codecademy (HTML)
* [JavaScript CheatSheet](https://htmlcheatsheet.com/js/) (HTML)
* [JavaScript Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/javascript) - Debuggex (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 5]
```diff
@@ -101,6 +102,11 @@
 * [jQuery CheatSheet](https://htmlcheatsheet.com/jquery/) (HTML)
 
 
+#### React.js
+
+* [React Cheatsheet](https://www.codecademy.com/learn/react-101/modules/react-101-jsx-u/cheatsheet) - Codecademy (React)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:107]: Tag use to mean the format or platform where the resource is writed, not what kind of resource is

```markdown
* [React Cheatsheet](https://www.codecademy.com/learn/react-101/modules/react-101-jsx-u/cheatsheet) - Codecademy (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 6]
```diff
@@ -134,15 +140,15 @@
 ### Python
 
 * [Comprehensive Python Cheat Sheet for Beginners](https://medium.com/the-codehub/comprehensive-python-cheat-sheet-for-beginners-5d76bb038fa2)
-* [Comprehensive Python Cheatsheet](https://gto76.github.io/python-cheatsheet) - Jure Šorn (HTML)
+* [Comprehensive Python Cheatsheet](https://gto76.github.io/python-cheatsheet) - Jure Šorn (Python)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:143]: Revert:
```markdown
* [Comprehensive Python Cheatsheet](https://gto76.github.io/python-cheatsheet) - Jure Šorn (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 7]
```diff
@@ -134,15 +140,15 @@
...
 * [Official Matplotlib cheat sheets](https://github.com/matplotlib/cheatsheets) - Matplotlib.org (LaTeX, PDF)
 * [Python Cheat Sheet](https://websitesetup.org/python-cheat-sheet/)
-* [Python Cheatsheet for beginners](https://www.codecademy.com/learn/learn-python-3/modules/learn-python3-syntax/cheatsheet) - Codeacademy (HTML)
+* [Python Cheatsheet for beginners](https://www.codecademy.com/learn/learn-python-3/modules/learn-python3-syntax/cheatsheet) - Codeacademy (Python)
 * [Python Crash Course Cheatsheet](https://ehmatthes.github.io/pcc/cheatsheets/README.html) - Eric Matthes (HTML)
 * [Python Data Wrangling with pandas](https://pandas.pydata.org/Pandas_Cheat_Sheet.pdf) - Pandas (PDF)
-* [Python for Data Science Cheatsheet](https://pydatascience.org/data-science-cheatsheets/) - DataCamp (HTML and Image)
-* [Python Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/python) - Debuggex (HTML)
+* [Python for Data Science Cheatsheet](https://pydatascience.org/data-science-cheatsheets/) - DataCamp (Python)
+* [Python Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/python) - Debuggex (Python)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:151]: Revert

```markdown
* [Python for Data Science Cheatsheet](https://pydatascience.org/data-science-cheatsheets/) - DataCamp (HTML and Image)
* [Python Regex Cheatsheet](https://www.debuggex.com/cheatsheet/regex/python) - Debuggex (HTML)
```

File: more/free-programming-cheatsheets.md [hunk 8]
```diff
@@ -134,15 +140,15 @@
 ### Python
 
 * [Comprehensive Python Cheat Sheet for Beginners](https://medium.com/the-codehub/comprehensive-python-cheat-sheet-for-beginners-5d76bb038fa2)
-* [Comprehensive Python Cheatsheet](https://gto76.github.io/python-cheatsheet) - Jure Šorn (HTML)
+* [Comprehensive Python Cheatsheet](https://gto76.github.io/python-cheatsheet) - Jure Šorn (Python)
 * [Learn Python in Y minutes](https://learnxinyminutes.com/docs/python/)
 * [Official Matplotlib cheat sheets](https://github.com/matplotlib/cheatsheets) - Matplotlib.org (LaTeX, PDF)
 * [Python Cheat Sheet](https://websitesetup.org/python-cheat-sheet/)
-* [Python Cheatsheet for beginners](https://www.codecademy.com/learn/learn-python-3/modules/learn-python3-syntax/cheatsheet) - Codeacademy (HTML)
+* [Python Cheatsheet for beginners](https://www.codecademy.com/learn/learn-python-3/modules/learn-python3-syntax/cheatsheet) - Codeacademy (Python)
```
Comment (davorpa) [inline more/free-programming-cheatsheets.md:147]: Revert 

```markdown
* [Python Cheatsheet for beginners](https://www.codecademy.com/learn/learn-python-3/modules/learn-python3-syntax/cheatsheet) - Codeacademy (HTML)
```

**Review by davorpa — CHANGES_REQUESTED** [review]
Tag use to mean the format or platform where the resource is stored, not what kind of resource is

Please, revert this changes

**Review by davorpa — APPROVED** [review]
Ready to merge :tada:

**Pull request merged**